src/lib.rs

```diff
@@ -748,4 +746,28 @@ Text (error at line 3):
 "
         );
     }
+
+    #[test]
+    fn test_allow_whitespace() {
+        let helper = |ptn: &str, text: &str| -> bool {
+            FMBuilder::new(ptn)
+                .unwrap()
+                .ignore_leading_whitespace(false)
+                .ignore_trailing_whitespace(false)
+                .build()
+                .unwrap()
+                .matches(text)
+                .is_ok()
+        };
+
+        assert!(helper("a\na", "a\na"));
+
+        assert!(helper("a\n a", "a\n a"));
+        assert!(!helper("a\n a", "a\na"));
+        assert!(!helper("a\na", "a\n a"));
+
+        assert!(helper("a\na ", "a\na "));
+        assert!(!helper("a\na", "a\na "));
+        assert!(!helper("a\na ", "a\na"));
```

Comment:
None of these test inputs have leading whitespace?

Or is it leading whitespace within each line?

Reply:
Within each line.

Reply:
e.g. this test:

```
        assert!(!helper("a\n a", "a\na"));
```

Reply:
I see. I think we should make that clearer that it's per-line. I thought we were talking about leading/trailing whitespace at the top/bottom of the entire test input.

Reply:
Agreed. Fixed in https://github.com/softdevteam/fm/pull/14/commits/7f2fac8568c0badfdf4f4bd359a6bb3f2565bc75.